builtin/common/vector.lua

```diff
@@ -375,6 +375,24 @@ function vector.in_area(pos, min, max)
 		(pos.z >= min.z) and (pos.z <= max.z)
 end
 
+function vector.random_direction()
+	-- Generate a random direction of unit length, via rejection sampling
+	local x = math.random(-1,1)
+    local y = math.random(-1,1)
+	local z = math.random(-1,1)
+    local l2 = x*x + y*y + z*z -- squared length
+	-- expected less than two attempts on average
+    while (l2 > 1 or l2 == 0) do -- rejected, retry
```

Comment:
This could be written as a `repeat`-`until` loop:

```lua
local x, y, z
local l2 -- squared length
repeat
    x, y, z = math.random() * 2 - 1, math.random() * 2 - 1, math.random() * 2 - 1
    l2 = x*x + y*y + z*z
until l2 >= 1e-6 and l2 <= 1 -- in unit sphere, not too close to origin
```

(Note that I'm also rejecting vectors close to the origin here to avoid (theoretical?) numeric issues)

Reply:
More elegant with a repeat-until indeed (haven't seen these since BASIC in the 80s, did not want while(true)).
Good point that the (rare) vectors close to the origin may end up being not very well chosen.